src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -669,13 +672,30 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
                 const auto cursorPosition = point.Position();
                 const auto terminalPosition = _GetTerminalPosition(cursorPosition);
 
-                // save location before rendering
-                _terminal->SetSelectionAnchor(terminalPosition);
-
                 // handle ALT key
                 _terminal->SetBoxSelection(altEnabled);
 
-                _renderer->TriggerSelection();
+                if (_IsTripleClick(cursorPosition, point.Timestamp()))

```

Comment:
Does this abstract well to `_NumberOfClicks(position, timestamp)`? Instead of adding another function for every number of clicks

Reply:
Take a look at this weird stuff.